tests/test_ramp_fitting.py

```diff
@@ -0,0 +1,408 @@
+
+
+def print_slope_poisson(slopes):
+    sdata, sdq, svp, svr, serr = slopes
+    print("Poisson:")
+    print(svp)
+
+
+def print_slope_readnoise(slopes):
+    sdata, sdq, svp, svr, serr = slopes
+    print("Readnoise:")
+    print(svr)
+
+
+def print_slope_err(slopes):
+    sdata, sdq, svp, svr, serr = slopes
+    print("Err:")
+    print(serr)
+
+
+def print_slopes(slopes):
+    print(DELIM)
+    print("**** SLOPES")
+    print(DELIM)
+    print_slope_data(slopes)
+
+    print(DELIM)
+    print_slope_poisson(slopes)
+
+    print(DELIM)
+    print_slope_readnoise(slopes)
+
+    print(DELIM)
+    print_slope_err(slopes)
+
+    print(DELIM)
+
+
+def print_integ_data(integ_info):
+    idata, idq, ivp, ivr, int_times, ierr = integ_info
+    print("Integration data:")
+    print(idata)
+
+
+def print_integ_poisson(integ_info):
+    idata, idq, ivp, ivr, int_times, ierr = integ_info
+    print("Integration Poisson:")
+    print(ivp)
+
+
+def print_integ_rnoise(integ_info):
+    idata, idq, ivp, ivr, int_times, ierr = integ_info
+    print("Integration read noise:")
+    print(ivr)
+
+
+def print_integ_err(integ_info):
+    idata, idq, ivp, ivr, int_times, ierr = integ_info
+    print("Integration err:")
+    print(ierr)
+
+
+def print_integ(integ_info):
+    print(DELIM)
+    print("**** INTEGRATIONS")
+    print(DELIM)
+    print_integ_data(integ_info)
+
+    print(DELIM)
+    print_integ_poisson(integ_info)
+
+    print(DELIM)
+    print_integ_rnoise(integ_info)
+
+    print(DELIM)
+    print_integ_err(integ_info)
+
+    print(DELIM)
+
+
+def print_optional_data(optional):
+    oslope, osigslope, ovar_poisson, ovar_rnoise, \
+        oyint, osigyint, opedestal, oweights, ocrmag = optional
+    print("Optional results slopes:")
+    print(f"Dimensions: {oslope.shape}")
+    print(oslope)
+
+
+def print_optional_poisson(optional):
+    oslope, osigslope, ovar_poisson, ovar_rnoise, \
+        oyint, osigyint, opedestal, oweights, ocrmag = optional
+    print("Optional results Poisson:")
+    print(f"Dimensions: {ovar_poisson.shape}")
+    print(ovar_poisson)
+
+
+def print_optional_rnoise(optional):
+    oslope, osigslope, ovar_poisson, ovar_rnoise, \
+        oyint, osigyint, opedestal, oweights, ocrmag = optional
+    print("Optional results read noise:")
+    print(f"Dimensions: {ovar_rnoise.shape}")
+    print(ovar_rnoise)
+
+
+def print_optional(optional):
+    print(DELIM)
+    print("**** OPTIONAL RESULTS")
+    print(DELIM)
+    print_optional_data(optional)
+
+    print(DELIM)
+    print_optional_poisson(optional)
+
+    print(DELIM)
+    print_optional_rnoise(optional)
+
+    print(DELIM)
+
+
+def print_all_info(slopes, cube, optional):
+    """
+    sdata, sdq, svp, svr, serr = slopes
+    idata, idq, ivp, ivr, int_times, ierr = cube
+    oslope, osigslope, ovp, ovr, \
+        oyint, osigyint, opedestal, oweights, ocrmag = optional
+    """
+
+    print(" ")
+    print_slopes(slopes)
+    print_integ(cube)
+    print_optional(optional)
```

Comment:
Do we really want all this info printed during test runs? What happens to the output when these are run in CI or Jenkins environments?

Reply:
These print functions are not run every time tests run.  They are not called in any test.  I use them for debugging and find myself rewriting them every time I write tests, so I kept them in the file this time.  I can remove them if you don't think they belong there.